server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSUsageTelemetry.java

```diff
@@ -8,172 +8,174 @@
+    private final LongMetric remotesPerSearch;
+    private final LongAdder skippedRemotes;
+
+    private final Map<String, LongAdder> featureCounts;
+
+    private final Map<String, LongAdder> clientCounts;
     private final Map<String, PerClusterCCSTelemetry> byRemoteCluster;
 
     public CCSUsageTelemetry() {
-        this.totalCCSCount = new LongAdder();
-        this.successfulSearchTelem = new SuccessfulCCSTelemetry();
-        this.failedSearchTelem = new FailedCCSTelemetry();
         this.byRemoteCluster = new ConcurrentHashMap<>();
+        totalCount = new LongAdder();
+        successCount = new LongAdder();
+        failureReasons = new ConcurrentHashMap<>();
+        took = new LongMetric();
+        tookMrtTrue = new LongMetric();
+        tookMrtFalse = new LongMetric();
+        remotesPerSearch = new LongMetric();
+        skippedRemotes = new LongAdder();
+        featureCounts = new ConcurrentHashMap<>();
+        clientCounts = new ConcurrentHashMap<>();
     }
 
     public void updateUsage(CCSUsage ccsUsage) {
+        // Ignore empty usage objects, we only care for true cross-cluster requests
+        if (ccsUsage.getRemotesCount() == 0) {
+            return;
+        }
         // TODO: fork this to a background thread? if yes, could just pass in the SearchResponse to parse it off the response thread
         doUpdate(ccsUsage);
     }
 
     // TODO: what is the best thread-safety model here? Start with locking model in order to get the functionality working.
     private synchronized void doUpdate(CCSUsage ccsUsage) {
-        totalCCSCount.increment();
-        if (ccsUsage.getFailureType() == null) {
-            // handle successful (or partially successful query)
-            successfulSearchTelem.update(ccsUsage);
-            for (Map.Entry<String, CCSUsage.PerClusterUsage> entry : ccsUsage.getPerClusterUsage().entrySet()) {
-                PerClusterCCSTelemetry perClusterCCSTelemetry = byRemoteCluster.get(entry.getKey());
-                if (perClusterCCSTelemetry == null) {
-                    perClusterCCSTelemetry = new PerClusterCCSTelemetry(entry.getKey());
-                    byRemoteCluster.put(entry.getKey(), perClusterCCSTelemetry);
-                }
-                // TODO: add more fields/data to the perClusterCCSTelemetry
-                perClusterCCSTelemetry.update(entry.getValue());
+        totalCount.increment();
+        long searchTook = ccsUsage.getTook();
+        if (isSuccess(ccsUsage)) {
+            successCount.increment();
+            // TODO: do we need to count latencies for failed requests? Right now the code doesn't collect it without SearchResponse
+            took.record(searchTook);
+            if (isMRT(ccsUsage)) {
+                tookMrtTrue.record(searchTook);
+            } else {
+                tookMrtFalse.record(searchTook);
             }
-
+            // TODO: can we do it on failure? Not sure what "took" values contain in that case if at all.
```

Comment:
Sometimes error responses will have took values, other times not. My vote is just to not record latencies for failure scenarios.